a10_octavia/controller/worker/flows/a10_member_flows.py

```diff
@@ -779,3 +779,240 @@ def get_create_member_snat_pool_subflow(self):
+                requires=[constants.MEMBER, a10constants.VTHUNDER, constants.POOL,
+                          constants.FLAVOR],
+                provides=a10constants.NAT_FLAVOR))
+            batch_update_members_flow.add(a10_database_tasks.GetNatPoolEntry(
+                name='get-nat-pool-entry-' + m.id,
+                inject={constants.MEMBER: m},
+                requires=[a10constants.NAT_FLAVOR],
+                provides=a10constants.NAT_POOL))
+            batch_update_members_flow.add(a10_network_tasks.ReleaseSubnetAddressForMember(
+                name='release-subnet-address-for-member-' + m.id,
+                inject={constants.MEMBER: m},
+                requires=[a10constants.NAT_FLAVOR, a10constants.NAT_POOL]))
+            batch_update_members_flow.add(a10_database_tasks.DeleteNatPoolEntry(
+                name='delete-nat-pool-entry-' + m.id,
+                requires=a10constants.NAT_POOL))
+            batch_update_members_flow.add(server_tasks.MemberDelete(
+                name='member-delete-' + m.id,
+                inject={constants.MEMBER: m},
+                requires=(constants.MEMBER, a10constants.VTHUNDER, constants.POOL,
+                          a10constants.MEMBER_COUNT_IP,
+                          a10constants.MEMBER_COUNT_IP_PORT_PROTOCOL)))
+            if CONF.a10_global.network_type == 'vlan':
+                batch_update_members_flow.add(
+                    vthunder_tasks.DeleteInterfaceTagIfNotInUseForMember(
+                        name='delete_unused_interface_tag_in_member_' + m.id,
+                        inject={constants.MEMBER: m},
+                        requires=[
+                            constants.MEMBER,
+                            a10constants.VTHUNDER]))
+            batch_update_members_flow.add(database_tasks.DeleteMemberInDB(
+                inject={constants.MEMBER: m},
+                name='{flow}-{id}'.format(
+                    id=m.id, flow=constants.DELETE_MEMBER_INDB)))
+            batch_update_members_flow.add(database_tasks.DecrementMemberQuota(
+                inject={constants.MEMBER: m},
+                name='{flow}-{id}'.format(
+                    id=m.id, flow=constants.DECREMENT_MEMBER_QUOTA_FLOW)))
+        batch_update_members_flow.add(
+            self.get_delete_member_vrid_internal_subflow(constants.POOL, old_members))
+
+        # for creation of members
+        batch_update_members_flow.add(lifecycle_tasks.MembersToErrorOnRevertTask(
+            name='{flow}-created'.format(
+                flow=constants.MEMBER_TO_ERROR_ON_REVERT_FLOW),
+            inject={constants.MEMBERS: new_members}))
+        for m in new_members:
+            batch_update_members_flow.add(database_tasks.MarkMemberPendingCreateInDB(
+                name='mark-member-pending-create-in-db-' + m.id,
+                inject={constants.MEMBER: m}))
+            batch_update_members_flow.add(a10_database_tasks.CountMembersWithIP(
```

Comment:
Can we optimize this? Like we could create a new task to get all the member ips and their counts. We store that into a dictionary. Then we lookup into the dictionary at a latter date instead of doing repeated (and in the worst case redundant) DB calls.

Reply:
Can you please explain how can we minimize the DB calls in this way?
Because we think evenif we create a new task to get all the member ips and their counts and store that into a dictionary, we will have to make a DB call for each member in the batch_update_member's query.